infra/hoogle_server.tf

```diff
@@ -196,8 +213,11 @@ resource "google_compute_backend_service" "hoogle-http" {
   health_checks = [google_compute_health_check.hoogle-http.self_link]
   port_name     = "http"
 
-  backend {
-    group = google_compute_instance_group_manager.hoogle.instance_group
+  dynamic backend {
```

Comment:
What happens for a dynamic backend if both have size /= 0? Does it do round-robin or something like that?

Reply:
This took a bit of research, but yes, I believe the final answer is round-robin across all machines (independent of group affiliation). At least that's my understanding of [this table](https://cloud.google.com/load-balancing/docs/features#load_balancing_methods), and the fact that we're using the EXTERNAL balancing scheme with the UTILIZATION balancing mode (with max=0.8).

So it would be round-robin amongst the machines that have CPU utilization below 0.8.

Reply:
Also note that this is not a "dynamic backend"; the `dynamic` here is a Terraform feature. This expands to:

```
resource "google_compute_backend_service" "hoogle-http" {
  name          = "hoogle-http"
  health_checks = [google_compute_health_check.hoogle-http.self_link]
  port_name     = "http"

  backend {
    group = google_compute_instance_group_manager.hoogle[0].instance_group
  }

  backend {
    group = google_compute_instance_group_manager.hoogle[1].instance_group
  }

}
```

So from Google's perspective this is a backend service with two backends.